File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;


vector<int> solution(vector<int> arr)
{
    arr.erase(min_element(arr.begin(), arr.end()));
    
    return !arr.size() ? vector<int>(1, -1) : arr;
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr) {
        int[] ints = new int[arr.length];
        int min = Integer.MAX_VALUE;
        for (int i : arr) {
            if (i < min) min = i;
        }
        int idx = 0;
        for (int i : arr) {
            if (i != min) {
                ints[idx++] = i;
            }
        }
        if (idx == 0) return new int[]{-1};
        return Arrays.copyOf(ints, idx);
    }
}